stdlib/time.pyi

```diff
@@ -32,39 +33,39 @@ if sys.version_info >= (3, 8) and sys.platform == "darwin":
 if sys.version_info >= (3, 9) and sys.platform == "linux":
     CLOCK_TAI: int
 
-class _struct_time(NamedTuple):
-    tm_year: int
-    tm_mon: int
-    tm_mday: int
-    tm_hour: int
-    tm_min: int
-    tm_sec: int
-    tm_wday: int
-    tm_yday: int
-    tm_isdst: int
+# The constructor of `struct_time` takes a sequence of any type, of length between 9 and 11 elements.
+# There is no way of expressing this in the type system.
+@final
+class struct_time(structseq[int]):
+    @overload  # type: ignore[override]
+    def __getitem__(self, __i: Literal[0, 1, 2, 3, 4, 5, 6, 7, 8]) -> int: ...
+    @overload
+    def __getitem__(self, __i: Literal[9, 10]) -> str: ...
+    @overload
+    def __getitem__(
+        self, __i: SupportsIndex
+    ) -> int: ...  # Strictly this should be `Any`, but `int` will be the majority of cases
```

Comment:
We could use `int | Any`, which won't error if you assume it's an int, but also won't cause unreachable code warnings if you use `assert isinstance(thing, str)`.

Reply:
If we do this, you should no longer get warnings about overloads with non-overlapping return types, and we should be able to remove `# type: ignore[misc]`.

Reply:
Good shout.